packages/frontend/src/utils/account-with-lockup.js

```diff
@@ -186,20 +213,19 @@ async function getAccountBalance(limitedAccountData = false) {
                 'get_account_total_balance', { account_id: lockupAccountId }));
             totalBalance = totalBalance.add(stakedBalanceLockup);
         }
-        const isFullyUnlocked = timeLeft.eq(new BN(0));
-        const ownersBalance = isFullyUnlocked
-            ? totalBalance.sub(new BN(terminationWithdrawnTokens))
-            : totalBalance.sub(BN.max(unreleasedAmount.sub(new BN(terminationWithdrawnTokens)), LOCKUP_MIN_BALANCE));
-
-        const lockedAmount = totalBalance.sub(ownersBalance);
-        const liquidOwnersBalance = BN.min(
-            ownersBalance, 
-            isFullyUnlocked 
-                ? stakedBalanceLockup.isZero()
-                    ? new BN(lockupBalance.total)
-                    : new BN(lockupBalance.total).sub(LOCKUP_MIN_BALANCE)
-                : new BN(lockupBalance.total).sub(LOCKUP_MIN_BALANCE)
-        );
+        
+        const ownersBalance = totalBalance.sub(lockedAmount)
+
+        // if acc is deletable (nothing locked && nothing stake) you can transfer the whole amount ohterwise get_liquid_owners_balance
+        const isAccDeletable = lockedAmount.isZero() && stakedBalanceLockup.isZero();
+        const liquidOwnersBalance = isAccDeletable
+            ? new BN(lockupBalance.total)
+            : new BN(
+                  await this.wrappedAccount.viewFunction(
+                      lockupAccountId,
+                      "get_liquid_owners_balance"
```

Comment:
Are we reasonably sure that `get_liquid_owners_balance` has always been part of the spec? :D

Reply:
We're using it [here](https://github.com/near/near-wallet/blob/master/packages/frontend/src/utils/account-with-lockup.js#L74) to transfer all available funds from the lockup. This makes sense because it's used by the `transfer` method [here](https://github.com/near/core-contracts/blob/master/lockup/src/owner.rs#L477) to determine the amount that is transferrable.

Reply:
@MaximusHaximus I don't have any lockups to test, but you can check with Patrick's at https://wallet.near.org/profile/patrick.near

I think I can get a legacy lockup up on Mainnet to check if on-chain and planned match, do you think is necessary?

Reply:
@stefanopepe I think Osman's been able to reliably test this at this point -- resolving :)